web/server.js

```diff
@@ -336,388 +322,345 @@ module.exports = function(options) {
-              oauthDb.generateAccessToken(
+            async method(request) {
+              return await oauthDb.generateAccessToken(
                 request.pre.client.client_id,
                 request.pre.authCode.user_id,
-                request.pre.authCode.scopes,
-                reply
+                request.pre.authCode.scopes
               );
             }
           }
         ]
       },
-      handler: function(request, reply) {
-        var responseObj = {
+      handler(request) {
+        return {
           access_token: request.pre.accessToken,
           scopes: request.pre.authCode.scopes,
           token_type: 'token'
         };
-
-        reply(responseObj);
       }
     },
     {
       method: 'POST',
       path: '/login',
-      config: {
+      options: {
         pre: [
           {
             assign: 'user',
-            method: function(request, reply) {
-              server.methods.account.verifyPassword(request, function(err, json) {
-                if ( err ) {
-                  return reply(err);
-                }
-
-                reply(json.user);
-              });
+            async method(request) {
+              const json = await server.methods.account.verifyPassword(request);
+
+              return json.user;
             }
           }
         ]
       },
-      handler: function(request, reply) {
-        request.auth.session.set(request.pre.user);
-        reply({ status: 'Logged In' });
+      handler(request) {
+        request.cookieAuth.set(request.pre.user);
+        return { status: 'Logged In' };
       }
     },
     {
       method: 'POST',
       path: '/request-reset',
-      config:{
+      options:{
         auth: false
       },
-      handler: function(request, reply) {
-        server.methods.account.requestReset(request, function(err, json) {
-          if ( err ) {
-            return reply(err);
-          }
-
-          reply(json);
-        });
+      async handler(request) {
+        return await server.methods.account.requestReset(request);
       }
     },
     {
       method: 'POST',
       path: '/reset-password',
-      config:{
+      options:{
         auth: false
       },
-      handler: function(request, reply) {
-        server.methods.account.resetPassword(request, function(err, json) {
-          if ( err ) {
-            return reply(err);
-          }
-
-          reply(json);
-        });
+      async handler(request) {
+        return await server.methods.account.resetPassword(request);
       }
     },
     {
       method: 'POST',
       path: '/create-user',
-      config: {
+      options: {
         auth: false,
         plugins: {
           crumb: false
         },
         cors: true,
         validate: {
           payload: {
-            username: Joi.string().regex(/^[a-zA-Z0-9\-]{1,20}$/).required(),
+            username: Joi.string().regex(/^[a-zA-Z0-9\-]{1,20}$/).required(), // eslint-disable-line no-useless-escape
```

Comment:
i'm not sure why eslint complains about escaping `-` inside the character class (I'm pretty sure it's necessary) but I didn't want to change it and possibly break something.